### ISSUE TITLE
[EXP] Add MeZO to Edge Pruning (FPT2InfoTrainer)

### ISSUE THREAD

**Issue by MustafaKhan0** [alansun17904/MAAA#7]
### Summary

Incorporate the edge-pruning repository / code into our repository. Override the current trainer with the MeZO training loop. Run initial experiments to confirm that our method works. Then run experiments on a larger GPU to compare our method to theirs in accuracy, runtime, and memory usage. 

### Method

- Determine if `Edge-Pruning/src` should be pasted into our repo or `Edge-Pruning` should be forked
- Paste `_inner_training_loop` from MeZO into FPT2InfoTrainer
- Add `zo_perturb_parameters`, `zo_forward`, `zo_forward_nondiff`, `zo_step`, `zo_update`, `_set_signature_columns_if_needed`, and `save_model`
- Check above functions to ensure they don't conflict with existing trainer implementation
- Add imports / dependencies
- Merge environments 
- Ensure that our implementation works (locally or Colab) - small, simple experiment, on gpt-2 (a small model)

### Computation

Local/Colab for initial evals

### Hypothesis

- The circuits should be quite similar / the same

### Acceptance Criteria

- MeZO is applied as an optimizer for the soft mask
- It does not take the same memory as a standards backwards pass
